testsuite/classlibrary/TestLinkClock.sc

```diff
@@ -202,4 +202,140 @@ TestLinkClock : UnitTest {
+				)
+			},
+			{
+				this.assert(false, "Make sure no other LinkClocks are running while testing");
+			}
+		);
+		clock1.stop;
+		clock2.stop;
+	}
+
+	// this test uses "magic number" wait statements
+	// because it takes a little time for a new LinkClock to stabilize
+	// and we have no signal to know when it's ready
+	test_LinkClock_sync_meter_aligns_barlines {
+		var numTrials = 5,
+		refClock, testClock, list = List.new, resp,
+		cond = Condition.new, outerCond = Condition.new;
+		refClock = LinkClock(10);
+		1.0.wait;  // refClock's time is not stable yet
+		refClock.schedAbs(4, Routine {
+			if(refClock.numPeers > 0) {
+				this.assert(false, "Make sure no other LinkClocks are running while testing");
+			} {
+				refClock.beatsPerBar = 3;
+				// we must wait until the reference clock finishes its initial sync attempt
+				resp = SimpleController(refClock).put(\resynced, {
+					resp.remove;
+					cond.unhang;
+				});
+				refClock.enableMeterSync;
+				cond.hang;
+				numTrials.do { |i|
+					if(i > 0) {  // don't wait first time
+						rrand(0.2, 3.0).wait;
+					};
+					testClock = LinkClock.new;
+					// wait to pick up the peer
+					resp = SimpleController(testClock).put(\numPeers, {
+						resp.remove;
+						cond.unhang;
+					});
+					cond.hang;
+					testClock.enableMeterSync;
+					resp = SimpleController(testClock).put(\meter, {
+						resp.remove;
+						cond.unhang;
+					});
+					cond.hang;
+					testClock.getMeterSync.queryMeter { |replies|
+						list.add(replies.every { |reply|
+							(reply[\queriedAtBeat] - testClock.baseBarBeat) % testClock.beatsPerBar
+							absdif: reply[\beatInBar] < 0.05
+						});
+						cond.unhang;
+					};
+					cond.hang;
+					testClock.stop;
+				};
+				this.assertEquals(
+					list.count { |bool| bool },
+					list.size,
+					"Count of successful trials should = number of trials"
+				);
+			};
+			outerCond.unhang;
+		});
+		outerCond.hang;
+		1.0.wait;
+		refClock.stop;
+	}
+
+	test_LinkClock_sync_meter_propagates_meter_changes {
+		var clock1 = LinkClock.new.enableMeterSync,
+		clock2, resp, cond = Condition.new, result;
+		1.0.wait;  // allow time for clock1 to find others
+		if(clock1.numPeers == 0) {
+			clock2 = LinkClock.new.enableMeterSync;
+			// wait for clock2 to finish initial sync
+			resp = SimpleController(clock2).put(\resynced, {
+				resp.remove;
+				cond.unhang;
+			});
+			cond.hang;
+			clock1.setMeterAtBeat(3, clock1.nextBar);
+			// wait for clock2 to sync meter again
+			resp = SimpleController(clock2).put(\meter, {
+				resp.remove;
+				cond.unhang;
+			});
+			cond.hang;
+			result = clock2.beatsPerBar == clock1.beatsPerBar;
+			this.assertEquals(clock2.beatsPerBar, clock1.beatsPerBar,
+				"clock1 and clock2 should have the same beatsPerBar after one changes"
+			);
+		} {
+			result = false;
+			this.assert(false, "Make sure no other LinkClocks are running while testing");
+		};
+		1.0.wait;
+		clock1.stop; clock2.stop;
```

Comment:
please put separate statements on separate lines. also, same question about the wait here.